Unused `type: ignore` error log coloring bug when using `show-error-codes`

* Are you reporting a bug, or opening a feature request?

Bug in log output coloring

* Please insert below the code you are checking with mypy

```python
def test(self, _):  # type: ignore[foo]
    pass
```

* What is the actual behavior/output?

Mypy correctly shows an error for the unused `type: ignore` comment, but the last letter ("t" from "comment") is colored yellow (where the error code would be shown). This does only happen with `--show-error-codes`.

```
$ mypy --warn-unused-ignores --show-error-codes test.py
```

![image](https://user-images.githubusercontent.com/19366641/71784937-6ec13100-2ff1-11ea-8091-b6bc41e163a4.png)

```
$ mypy --warn-unused-ignores test.py
```

![image](https://user-images.githubusercontent.com/19366641/71784998-500f6a00-2ff2-11ea-8ce8-2f7d5e0b65e8.png)

* What is the behavior/output you expect?

Last character shouldn't be colored OR mypy should show an error code here as well, something like:

```
test.py:1: error: unused 'type: ignore' comment  [unused-type-ignore]
```

Not sure if that makes sense as this code couldn't be used in a `type: ignore` comment to... ignore a `type: ignore` comment...

* What are the versions of mypy and Python you are using?
  Do you see the same issue after installing mypy from Git master?

```
$ python -V
Python 3.7.5
$ mypy -V
mypy 0.761
```

Same happens with mypy installed from git master:

```
$ mypy -V
mypy 0.770+dev.cdd91ba819646b0138b05a8e9180faba4ef0ff92
```

* What are the mypy flags you are using? (For example --strict-optional)

`--warn-unused-ignores` and `--show-error-codes`

--- a/mypy/util.py
+++ b/mypy/util.py
@@ -607,8 +607,11 @@ class FancyFormatter:
                 return (loc + self.style('error:', 'red', bold=True) +
                         self.highlight_quote_groups(msg))
             codepos = msg.rfind('[')
-            code = msg[codepos:]
-            msg = msg[:codepos]
+            if codepos != -1:
+                code = msg[codepos:]
+                msg = msg[:codepos]
+            else:
+                code = ""  # no error code specified
             return (loc + self.style('error:', 'red', bold=True) +
                     self.highlight_quote_groups(msg) + self.style(code, 'yellow'))
         elif ': note:' in error:

